Library/Homebrew/utils/curl.rb

```diff
@@ -1,42 +1,38 @@
 require "pathname"
 require "open3"
 
-def curl_args(options = {})
+def curl_executable
   curl = Pathname.new ENV["HOMEBREW_CURL"]
   curl = Pathname.new "/usr/bin/curl" unless curl.exist?
-  raise "#{curl} is not executable" unless curl.exist? && curl.executable?
+  return curl if curl.executable?
+  raise "#{curl} is not executable"
+end
+
+def curl_args(*extra_args, show_output: false, user_agent: HOMEBREW_USER_AGENT_CURL)
+  args = [curl_executable.to_s]
 
-  args = [
-    curl.to_s,
-    "--remote-time",
-    "--location",
-  ]
+  args << "--remote-time" unless extra_args.each_cons(2).include?(["--output", "/dev/null"])
 
-  case options[:user_agent]
-  when :browser
+  if user_agent == :browser
     args << "--user-agent" << HOMEBREW_USER_AGENT_FAKE_SAFARI
   else
-    args << "--user-agent" << HOMEBREW_USER_AGENT_CURL
+    args << "--user-agent" << user_agent
   end
 
-  unless options[:show_output]
+  unless show_output
     args << "--progress-bar" unless ARGV.verbose?
     args << "--verbose" if ENV["HOMEBREW_CURL_VERBOSE"]
     args << "--fail"
     args << "--silent" if !$stdout.tty? || ENV["TRAVIS"]
   end
 
-  args += options[:extra_args] if options[:extra_args]
-  args
+  args + extra_args
 end
 
 def curl(*args)
-  safe_system(*curl_args(extra_args: args))
+  safe_system(*curl_args("--location", *args))
```

Comment:
Ok but it's not clear from which which commands and why it's present on `curl` and not `curl_args`.